src/rpc_client/src/lib.rs

```diff
@@ -84,12 +79,7 @@ pub trait RpcClient: Send + Sync + 'static + Clone {
 pub struct RpcClientPool<S> {
     connection_pool_size: u16,
 
-    #[cfg(not(madsim))]
```

Comment:
Tried to enable Moka cache under `madsim` according to the change log. cc @wangrunji0408 

> No more background threads: All cache types future::Cache, sync::Cache, and sync::SegmentedCache no longer spawn background threads.
>
> The scheduled-thread-pool crate was removed from the dependency.
Because of this change, many private methods and some public methods under the future module were converted to async methods. You may need to add .await to your code for those methods.